src/jit/lowerarmarch.cpp

```diff
@@ -121,6 +121,7 @@ bool Lowering::IsContainableImmed(GenTree* parentNode, GenTree* childNode)
             case GT_LE:
             case GT_GE:
             case GT_GT:
+            case GT_JCMP:
                 return emitter::emitIns_valid_imm_for_cmp(immVal, size);
```

Comment:
This call to `emitIns_valid_imm_for_cmp` isn't correct for these instrutions:

A specialized version needs to be added here, that allows for an immediate of 0 or an exact  power of two

Reply:
Done